frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -454,9 +470,11 @@ export default function WaveformGraph({
       .data(
         initialPeaksValleys[0].filter((peak) => {
           const isPeakWithinWindow = dataToGraph[peak][0] >= startTime && dataToGraph[peak][0] <= endTime;
-          return peakValleyWindows[currentWell].minPeaks
-            ? isPeakWithinWindow && dataToGraph[peak][1] >= peakValleyWindows[currentWell].minPeaks
-            : isPeakWithinWindow;
+          const actualY = dataToGraph[peak][1];
+          const computedY =
+            dataToGraph[peak][0] * peakSlope[wellNameToIndex[currentWell]] +
```

Comment:
I think peak slope should just be calculated as needed since we have enough information to do so and it's not an expensive calculation

Reply:
I do think peak slope could be calculated, but I think storing it makes the code less confusing. I can make the change but I would like to leave it because it makes the code easier to read.

Reply:
Why does it make the code less confusing? I don't think the slope is necessary in any other parts of the code